public/public.php

```diff
@@ -467,30 +467,33 @@ public function list_post_attachments_with_sources( $post_id = 0 ) {
 
 		if ( ! empty( $attachments ) ) {
 			ISC_Log::log( sprintf( 'going through %d attachments', count( $attachments ) ) );
-			$atts = array();
+			$atts = [];
 			foreach ( $attachments as $attachment_id => $attachment_array ) {
 				$use_standard_source = Standard_Source::use_standard_source( $attachment_id );
 				$source              = self::get_image_source_text( $attachment_id );
 
 				// check if source of own images can be displayed
-				if ( ( $use_standard_source == '' && $source == '' ) || ( $use_standard_source != '' && $exclude_standard ) ) {
-					if ( $use_standard_source != '' && $exclude_standard ) {
+				if ( ( $use_standard_source === '' && $source === '' ) || ( $use_standard_source !== '' && $exclude_standard ) ) {
+					if ( $use_standard_source !== '' && $exclude_standard ) {
 						ISC_Log::log( sprintf( 'image %d: "own" sources are excluded', $attachment_id ) );
 					} else {
 						ISC_Log::log( sprintf( 'image %d: skipped because of empty source', $attachment_id ) );
 					}
 					unset( $atts[ $attachment_id ] );
-					continue;
 				} else {
 					$atts[ $attachment_id ]['title'] = get_the_title( $attachment_id );
 					ISC_Log::log( sprintf( 'image %d: getting title "%s"', $attachment_id, $atts[ $attachment_id ]['title'] ) );
 					$atts[ $attachment_id ]['source'] = $this->render_image_source_string( $attachment_id );
+					if ( ! $atts[ $attachment_id ]['source'] ) {
+						ISC_Log::log( sprintf( 'image %d: skipped because of empty standard source', $attachment_id ) );
+						unset( $atts[ $attachment_id ] );
+					}
```

Comment:
**Suggestion:** It's recommended to use `empty()` function to check if a variable is empty or not. This function will return `true` if the variable is an empty string, false, array(), NULL, “0”, 0, and an unset variable. [best practice]
```suggestion
     if ( empty($atts[ $attachment_id ]['source']) ) {
       ISC_Log::log( sprintf( 'image %d: skipped because of empty standard source', $attachment_id ) );
       unset( $atts[ $attachment_id ] );
     }
```